### PR TITLE
removing strip-comments and using prettier to clean pragma versions

### PR DESCRIPTION
**Pull request by Janther**
Aware this might be an overkill but the possible issue is really difficult to solve with other means (see tests).

I have a different approach to get rid of the `strip-comments` dependencies if this gets rejected.

### CODE REVIEW

File: src/slang-utils/create-parser.ts [hunk 1]
```diff
@@ -25,15 +32,48 @@ const milestoneVersions = Array.from(
...
+          locEnd
+        } as PrettierParser
+      },
+      printers: { ['slang-ast']: { print: slangPrint } }
+    }
+  ],
+  parser: 'slangPragma'
+};
+
+const query = Query.parse(
```
Comment (mattiaerre) [inline src/slang-utils/create-parser.ts:64]: I noticed you moved `query` out of the `tryToCollectPragmas` function, but you only use it there; should we move it back to its original location?

File: tests/unit/slang-utils/create-parser.test.js [hunk 2]
```diff
@@ -83,21 +83,31 @@ describe('inferLanguage', function () {
     {
       description:
         'should use the latest version if the range is outside the supported versions',
-      source: `pragma solidity ^0.8.27;`,
+      source: `pragma solidity ^0.9.27;`,
       version: latestSupportedVersion
+    },
+    {
+      description: 'broken by new lines, whitespace and comments',
```
Comment (mattiaerre) [inline tests/unit/slang-utils/create-parser.test.js:90]:
```suggestion
      description: 'broken by new lines, whitespaces, and comments',
```

**Review by mattiaerre — APPROVED** [review]

File: src/slang-utils/create-parser.ts (hunk 1 above)
Comment (Janther) [reply src/slang-utils/create-parser.ts:64]: this way the query is computed only once and not every time it tryToCollectPragmas is called

**Review by mattiaerre — APPROVED** [review]
🚀

**Comment by fvictorio** [comment]
I think I'd rather not support very weird pragma versions, like the one in the new test, and rather check that we throw a decent error when that happens (throw "couldn't parse version pragma in file <file>" and crash).

**Comment by Janther** [comment]
I made a new PR I had prepared if this one was too ambitious and complicated maintenance

**Pull request closed without merging**